src/pages/permissions/panels/CreateGroupPanel.tsx

```diff
@@ -60,37 +104,102 @@ const CreateGroupPanel: FC = () => {
 
   return (
     <>
-      <SidePanel isOverlay loading={false} hasError={false}>
+      <SidePanel
+        isOverlay
+        loading={false}
+        hasError={false}
+        className={classnames({
+          "edit-permissions-panel": subForm === "permission",
+        })}
+      >
         <SidePanel.Header>
-          <SidePanel.HeaderTitle>Create group</SidePanel.HeaderTitle>
+          {subForm === null && (
+            <SidePanel.HeaderTitle>Create group</SidePanel.HeaderTitle>
+          )}
+          {subForm === "identity" && (
+            <SidePanel.HeaderTitle>
+              <Icon name="chevron-left" />{" "}
+              <Button
+                onClick={() => setSubForm(null)}
+                dense
+                appearance="link"
+                className="p-heading--4"
+              >
+                Create group
+              </Button>{" "}
+              / Add identities
+            </SidePanel.HeaderTitle>
+          )}
+          {subForm === "permission" && (
+            <SidePanel.HeaderTitle>
+              <Icon name="chevron-left" />{" "}
+              <Button
+                onClick={() => setSubForm(null)}
+                dense
+                appearance="link"
+                className="p-heading--4"
+              >
+                Create group
+              </Button>{" "}
+              / Add permissions
+            </SidePanel.HeaderTitle>
+          )}
         </SidePanel.Header>
         <NotificationRow className="u-no-padding" />
         <SidePanel.Content className="u-no-padding">
           <ScrollableContainer
-            dependencies={[notify.notification]}
+            dependencies={[notify.notification, subForm]}
             belowIds={["panel-footer"]}
           >
-            <GroupForm formik={formik} />
+            {subForm === null && (
+              <GroupForm
+                formik={formik}
+                setSubForm={setSubForm}
+                identityCount={identities.length}
+                identityModifyCount={identities.length}
+                permissionCount={permissions.filter((p) => !p.isRemoved).length}
```

Comment:
With the new logic of removing those on creation instead of setting them as removed, this simplifies to not need a filter. similar to the identitites.length above.